pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/UnpublishVersion.java

```diff
@@ -88,15 +91,19 @@ public void run(SlingHttpServletRequest request, PostResponse response, SlingPos
         if (response.getError() == null) {
             // call the extension point
             Locale locale = getLocale(request);
-            Module module = getModule(request);
+            Optional<? extends Document> document = getDocument(request, request.getResourceResolver());
             String variant = getVariant(request);
-            ModuleVersion moduleVersion = module.locale(locale).get()
+            Optional<? extends DocumentVersion> documentVersion = Optional.of(document.get().locale(locale).get()
```

Comment:
Same comments here, I am betting this `Optional` construct is unnecessary.

Reply:
@benradey we have below method structure for method getReleasedVersion(), so DocumentVersion instance wont be able to call below method in case we remove optional construct.

default Optional<? extends DocumentVersion> getReleasedVersion(@Nonnull final Locale locale,@Nonnull final String variantName) {